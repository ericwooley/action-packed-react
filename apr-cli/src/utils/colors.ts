import chalk from "chalk";

export const { green, grey, blue, red } = chalk;
