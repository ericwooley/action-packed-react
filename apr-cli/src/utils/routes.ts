import debug from "debug";
import glob from "glob";
import { join, sep } from "path";
import fs from "fs";

export function findRoutes(name: string, { pretty = false } = {}) {
  const log = debug("apr:generate:" + name);
  const dir = join(process.cwd(), "/**/routes/*");

  log("scanning: ", dir);
  let routes = glob.sync(dir);
  log("unprocessed routes", routes, sep);
  routes = routes
    .filter(file => fs.statSync(file).isDirectory())
    .filter(file => file !== "." && file !== "..")
    .map(route => {
      const formatted = route
        .split(sep)

        .join(sep)
        .replace(`${process.cwd()}${sep}`, "");
      if (pretty) return formatted.replace(new RegExp(`${sep}routes${sep}`, "g"), " -> ");
      return formatted;
    });
  log("routes: ", routes);
  return ["src/", ...routes];
}

const routeStr = `${sep}routes${sep}`;
const prettySep = " -> ";
const routeSepRGX = new RegExp(routeStr, "g");

export function prettyRoute(route: string, {newSep = prettySep} = {}) {
  return route.replace(routeSepRGX, newSep).replace(/^src\//, '/');
}
export function uglyRoute(route: string, {oldSep = prettySep} = {}) {
  return route.replace(new RegExp(oldSep, "g"), routeStr).replace(/^\//, 'src\//');
}
