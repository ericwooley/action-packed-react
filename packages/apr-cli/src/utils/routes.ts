import debug from "debug";
import glob from "glob";
import { join, sep } from "path";
import fs from "fs";
const routeLog = debug("apr:foundRoutes")
export function findRoutes(name: string, { pretty = false } = {}) {
  const log = debug("apr:generate:" + name);
  const dir = join(process.cwd(), "./**/routes");

  log("scanning: ", dir);
  let routes = glob.sync(dir);
  log("unprocessed routes", routes, sep);
  routes = routes
    .filter(file => fs.statSync(file).isDirectory())
    .filter(file => file !== "." && file !== "..")
    .map(route => {
      const formatted = route
      .replace(new RegExp(`${sep}routes$`), '')
      .split(sep)
      .join(sep)
      .replace(`${process.cwd()}${sep}`, "")
      // glob bug????
        .replace(/^srcroutes/, `src${sep}routes`);
      if (pretty) return prettyRoute(formatted);
      return formatted;
    });
  log("routes: ", routes);
  return [...routes];
}

const routeStr = `${sep}routes${sep}`;
const routeSepRGX = new RegExp(routeStr, "g");
const prettySep = " -> ";
routeLog("routeStr", routeStr)

export function prettyRoute(route: string, { newSep = prettySep } = {}) {
  const pretty = route.replace(routeSepRGX, newSep);
  routeLog("pretty -", route, ': |', pretty, '| sep=', prettySep)
  return pretty
}
export function uglyRoute(route: string, { oldSep = prettySep } = {}) {
  const ugly = route.replace(new RegExp(oldSep, "g"), routeStr);
  routeLog(route, ': |', ugly, '| sep=', prettySep)
  return ugly
}
