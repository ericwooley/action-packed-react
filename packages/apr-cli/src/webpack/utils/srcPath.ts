const path = require('path')
module.exports = function srcPath (subdir: string) {
  const result = path.join(process.cwd(), 'src', subdir)
  return result
}
