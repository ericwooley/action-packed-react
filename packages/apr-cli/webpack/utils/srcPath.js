const path = require('path')
module.exports = function srcPath (subdir) {
  const result = path.join(process.cwd(), 'src', subdir)
  return result
}
