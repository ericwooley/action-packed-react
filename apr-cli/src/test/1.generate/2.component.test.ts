import { snapshotPlayground } from "../utils/getAllSourceFiles";
import { runApr } from "../utils/runApr";
describe("generating", () => {
  describe("component", () => {
    it("should generate a new component", async () => {
      runApr("g component --name testComponent --route src", { snapshotOutput: true });
      await snapshotPlayground();
    });
    it("should generate a new component based on testRoutes", async () => {
      runApr("g component --name testComponent2 --route src/routes/testRoute", {
        snapshotOutput: true
      });
      await snapshotPlayground();
    });
    it("should generate a new component based on testRoutes2", async () => {
      runApr("g component --name testComponent3 --route src/routes/testRoute/routes/testRoute2", {
        snapshotOutput: true
      });
      await snapshotPlayground();
    });
    it("should throw an error on a non-existent route", () => {
      expect(() =>
        runApr("g component --name testComponent --route src/routes/nothing")
      ).toThrowErrorMatchingInlineSnapshot(
        `"Unsuccessful apr command: apr g component --name testComponent --route src/routes/nothing"`
      );
    });
  });
});
