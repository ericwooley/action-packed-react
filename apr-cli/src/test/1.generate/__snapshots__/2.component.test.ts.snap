// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generating component should generate a new component: apr g component --name testComponent --route src --autoLink 1`] = `
",
Loaded templates: <project-root>apr-cli/_templates/
[32m       added: src/components/testComponent/index.ts[39m
[32m       added: src/components/testComponent/testComponent.tsx[39m
[32m       added: src/components/testComponent/testComponent.story.tsx[39m
,"
`;

exports[`generating component should generate a new component: file-list 1`] = `
Array [
  "<project-root>playground/src/components/testComponent/index.ts",
  "<project-root>playground/src/components/testComponent/testComponent.story.tsx",
  "<project-root>playground/src/components/testComponent/testComponent.tsx",
]
`;
